### ISSUE TITLE
Some questions about SAM

### ISSUE THREAD

**Issue by DUT-CSJ** [Scatteredrain/SALI#6]
感谢分享这么棒的工作！
关于SAM我有几个问题，
1）论文中提到offset是用1x1卷积生成的，但是我看到代码里写的似乎是3x3卷积
2）论文中提到将两帧特征图直接拼起来作为生成offset的条件，但如果两帧的空间位移较大，直接拼起来在空间上并不是对齐的，用这样的条件生成offset似乎在原理上解释不通
作者能否解答一下以上问题呢？无比感谢！

**Comment by Huster-Hq** [comment]
Thank you for your interest in our work.

Below is a response to your question:
1. You are right, we may have some slight differences in the code implementation from the paper writing.
2. This approach makes sense and has been adopted by a lot of works, and we adopted this approach inspired by the work[1].

Hope the above answer can addresses your question.

Ref:
[1] EDVR: Video Restoration with Enhanced Deformable Convolutional Networks, 2019.

**Comment by Huster-Hq** [comment]
> 感谢分享这么棒的工作！ 关于SAM我有几个问题， 1）论文中提到offset是用1x1卷积生成的，但是我看到代码里写的似乎是3x3卷积 2）论文中提到将两帧特征图直接拼起来作为生成offset的条件，但如果两帧的空间位移较大，直接拼起来在空间上并不是对齐的，用这样的条件生成offset似乎在原理上解释不通 作者能否解答一下以上问题呢？无比感谢！

In response to the second question, this implementation is perhaps suboptimal, and you can use a more superior algorithm to replace it; if you have a better concise one, feel free to share it with us.

**Comment by DUT-CSJ** [comment]
> Thank you for your interest in our work.
> 
> Below is a response to your question:
> 
> 1. You are right, we may have some slight differences in the code implementation from the paper writing.
> 2. This approach makes sense and has been adopted by a lot of works, and we adopted this approach inspired by the work[1].
> 
> Hope the above answer can addresses your question.
> 
> Ref: [1] EDVR: Video Restoration with Enhanced Deformable Convolutional Networks, 2019.

Thank you for your reply!
There are some other question:
1. The batch size (8) and weight decay (0) in your code is different from the paper (14 and 1e-4). For a 24GB GPU, the maximum of batch size is 8.
2. The paper does not provide any information regarding the setup of the skip_list. I find that the best Dice score appear in the first few epochs. Does the skip_list have an impact on performance? According to the skip_list settings in the code, the model I train achieves a lower Dice score compared to the weights you provide.

**Comment by Huster-Hq** [comment]
> > Thank you for your interest in our work.
> > Below is a response to your question:
> > 
> > 1. You are right, we may have some slight differences in the code implementation from the paper writing.
> > 2. This approach makes sense and has been adopted by a lot of works, and we adopted this approach inspired by the work[1].
> > 
> > Hope the above answer can addresses your question.
> > Ref: [1] EDVR: Video Restoration with Enhanced Deformable Convolutional Networks, 2019.
> 
> Thank you for your reply! There are some other question:
> 
> 1. The batch size (8) and weight decay (0) in your code is different from the paper (14 and 1e-4). For a 24GB GPU, the maximum of batch size is 8.
> 2. The paper does not provide any information regarding the setup of the skip_list. I find that the best Dice score appear in the first few epochs. Does the skip_list have an impact on performance? According to the skip_list settings in the code, the model I train achieves a lower Dice score compared to the weights you provide.

**Comment by Huster-Hq** [comment]
> > Thank you for your interest in our work.
> > Below is a response to your question:
> > 
> > 1. You are right, we may have some slight differences in the code implementation from the paper writing.
> > 2. This approach makes sense and has been adopted by a lot of works, and we adopted this approach inspired by the work[1].
> > 
> > Hope the above answer can addresses your question.
> > Ref: [1] EDVR: Video Restoration with Enhanced Deformable Convolutional Networks, 2019.
> 
> Thank you for your reply! There are some other question:
> 
> 1. The batch size (8) and weight decay (0) in your code is different from the paper (14 and 1e-4). For a 24GB GPU, the maximum of batch size is 8.
> 2. The paper does not provide any information regarding the setup of the skip_list. I find that the best Dice score appear in the first few epochs. Does the skip_list have an impact on performance? According to the skip_list settings in the code, the model I train achieves a lower Dice score compared to the weights you provide.

1. Yes, there are minor discrepancies between the code posted on Github and some of the details written in the paper, and when there is a contradiction, we recommend that the code posted on Github prevails, e.g., training epoch, batch size, etc.
2. We set _skip_list_ to enrich the diversity of training set construction, which is not part of the core of the methodology in this paper and therefore not mentioned in the text. Although the original intention is good, it is found through experiments that _skip_list_ is not helpful for performance improvement.
3. The results presented in our paper were run through this code without the slightest alteration. As for the lower Dice score you achieved, it may be due to the randomness of the training, and we suggest that you can run it a few more times. You can share the best results you've run so far and see how close they are to the results published in the paper.

**Comment by DUT-CSJ** [comment]
> > > Thank you for your interest in our work.
> > > Below is a response to your question:
> > > 
> > > 1. You are right, we may have some slight differences in the code implementation from the paper writing.
> > > 2. This approach makes sense and has been adopted by a lot of works, and we adopted this approach inspired by the work[1].
> > > 
> > > Hope the above answer can addresses your question.
> > > Ref: [1] EDVR: Video Restoration with Enhanced Deformable Convolutional Networks, 2019.
> > 
> > 
> > Thank you for your reply! There are some other question:
> > 
> > 1. The batch size (8) and weight decay (0) in your code is different from the paper (14 and 1e-4). For a 24GB GPU, the maximum of batch size is 8.
> > 2. The paper does not provide any information regarding the setup of the skip_list. I find that the best Dice score appear in the first few epochs. Does the skip_list have an impact on performance? According to the skip_list settings in the code, the model I train achieves a lower Dice score compared to the weights you provide.
> 
> 1. Yes, there are minor discrepancies between the code posted on Github and some of the details written in the paper, and when there is a contradiction, we recommend that the code posted on Github prevails, e.g., training epoch, batch size, etc.
> 2. We set _skip_list_ to enrich the diversity of training set construction, which is not part of the core of the methodology in this paper and therefore not mentioned in the text. Although the original intention is good, it is found through experiments that _skip_list_ is not helpful for performance improvement.
> 3. The results presented in our paper were run through this code without the slightest alteration. As for the lower Dice score you achieved, it may be due to the randomness of the training, and we suggest that you can run it a few more times. You can share the best results you've run so far and see how close they are to the results published in the paper.

Thank you for your detailed answer!
